packages/teuchos/parameterlist/src/Teuchos_VerboseObjectParameterListHelpers.cpp

```diff
@@ -117,19 +117,81 @@ void Teuchos::readVerboseObjectSublist(
   *verbLevel = VerbosityLevel_validator->getIntegralValue(
     voSublist,VerbosityLevel_name,VerbosityLevel_default
     );
+  // the default file string is nothing
   if (outputFileStr==OutputFile_default) {
     *oStream = null;
   }
+  // if a file is specified then output to an fstream
   else {
-    RCP<std::ofstream>
-      oFileStream = rcp(new std::ofstream(outputFileStr.c_str()));
-    TEUCHOS_TEST_FOR_EXCEPTION_PURE_MSG(
-      oFileStream->eof(), Exceptions::InvalidParameterValue,
-      "Error, the file \"" << outputFileStr << "\n given by the parameter\n"
-      "\'" << OutputFile_name << "\' in the sublist\n"
-      "\'" << voSublist.name() << "\' count not be opened for output!"
-      );
-    *oStream = fancyOStream(rcp_implicit_cast<std::ostream>(oFileStream));
+
+    // JJE: 14 March 2019
+    //  A fix for file output of an VerboseObject.
+    //
+    // This step is very important. With filestreams it does not make
+    // sense for multiple MPI ranks to open the same file.  Nor,
+    // does it seem inline with the OStream model that each stream
+    // represent a unique file. Perhaps, if that functionality is desired
+    // then the file name could be suffixed with the MPI Rank.
+    //
+    // A fundamental flaw with this implementation is that we have no knowledge
+    // of a communicator on which this OStream belongs. That makes the idea
+    // of using a rank ambiguous.
+    //
+    // The code below was added, and uses COMM_WORLD, because as-is
+    // this functionality was fundamentally broken. Without restricting
+    // the stream to a single rank, two severe consquences follow:
+    //   1) Each MPI process opens the file, which is not scalable;
+    //   2) Moreover, each MPI process *writes* to the file. Which
+    //      can give the illusion that things are working, if each
+    //      process writes exactly the same information (e.g., solver
+    //      convergence information for a bulk synchronous solve).
+    //      This introduces a terrible scalability problem, as the
+    //      filesystem is then tasked with coping with concurrent writes
+    //      to the same shared file, which is should make you cry a little.
+    //
+    // The resolution, is two fold:
+    //   1st, construct the ostream as a regular wrapper around cout
+    //   2nd, restrict the file creation and opening to a single process
+    //   Finally, map all ostreams except the fstream one to
+    //    a blackhole. Ensuring each rank has a functional stream
+    //    but that only one actually emits data to disk
+    //
+
+    // this could be a BlackHole, but calling setOutputToRootOnly does slightly
+    // more than simply blackhole output, it also disabled buffering across processes
+    *oStream = Teuchos::fancyOStream(Teuchos::rcpFromRef(std::cout));
```

Comment:
This is safe since `std::cout` is guaranteed to live past all possible references, even after `main()` ends.